x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportOpenJobAction.java

```diff
@@ -236,6 +240,16 @@ static void validate(String jobId, Job job) {
                     reasons.add(reason);
                     continue;
                 }
+
+                boolean jobConfigIsStoredInIndex = job.getJobVersion().onOrAfter(Version.V_6_6_0);
+                if (jobConfigIsStoredInIndex && node.getVersion().before(Version.V_6_6_0)) {
+                    String reason = "Not opening job [" + jobId + "] on node [" + nodeNameOrId(node)
+                            + "] version [" + node.getVersion() + "], because this node does not support " +
+                            "jobs of version [" + job.getJobVersion() + "]";
+                    logger.trace(reason);
```

Comment:
nit: debug seems more suited to me

Reply:
Remember that on a 100 node cluster each allocation will generate 100 messages similar to this one, which would be significant log spam.  They get concatenated into the overall reason, which is stored in the cluster state if the persistent task exists (and returned in the error message in the case of this being called prior to opening).

All the other possible reasons for ruling out a node in this method also currently log at the trace level.  I think they should all log at the same level, otherwise someone reading the logs could get a misleading picture of what is happening.

I would leave this as trace to match the others.